matrix-connection.c

```diff
@@ -443,10 +443,10 @@ void matrix_connection_join_room(struct _PurpleConnection *pc,
     /* we have to copy the components table, so that we can pass it back
      * later on :/
      */
-    copy = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
```

Comment:
Shouldn't this remain as the `g_hash_table_new_full` to prevent use-after-free and other memory issues?

Reply:
../pidgin-2.14.10/libpurple/roomlist.c check purple_roomlist_room_join() g_hash_table_new.

It is for a reason.

room can handle two types of data: int and string. you can't g_free on int - crash.